File: yizhi/Alien-V4-improved-master/src/main/java/dev/gzsakura_miitong/api/utils/math/Timer.java
/*
 * Decompiled with CFR 0.152.
 */
package dev.gzsakura_miitong.api.utils.math;

public class Timer {
    private long startTime = -1L;

    public Timer() {
        this.reset();
    }

    public void reset() {
        this.startTime = System.currentTimeMillis();
    }

    public long getMs() {
        return System.currentTimeMillis() - this.startTime;
    }

    public boolean passedS(double s) {
        return this.passed((long)s * 1000L);
    }

    public boolean passedMs(double ms) {
        return this.passed((long)ms);
    }

    public void setMs(long ms) {
        this.startTime = System.currentTimeMillis() - ms;
    }

    public boolean passed(long ms) {
        return System.currentTimeMillis() - this.startTime >= ms;
    }
}

File: src/main/kotlin/com/zychen027/MeteorPlusPlus/utils/math/Timer.kt
package com.zychen027.meteorplusplus.utils.math

class Timer {
    private var time = -1L

    fun reset() {
        time = System.currentTimeMillis()
    }

    fun setMs(ms: Long) {
        time = System.currentTimeMillis() - ms
    }

    fun passedMs(ms: Long): Boolean {
        return System.currentTimeMillis() - time >= ms
    }

    fun getMs(): Long {
        return System.currentTimeMillis() - time
    }
}
